src/components/dialog/dialog.js

```diff
@@ -336,6 +342,18 @@ function MdDialogProvider($$interimElementProvider) {
         element.css('top', parentOffset +'px');
       }
 
+      var role = 'dialog',
+          elementToFocus = closeButton;
+
+      if (options.$type === 'alert') {
+        role = 'alertdialog';
+        elementToFocus = element.find('md-content');
+      }
+
+      configureAria(element.find('md-dialog'), role);
+
+      document.addEventListener('focus', trapFocus, true);
```

Comment:
The native event listener was required to enable the capture phase. Relevant source (see "Trapping focus"): http://www.nczonline.net/blog/2013/02/12/making-an-accessible-dialog-box/